sys/riscv/riscv/unwind.c

```diff
@@ -50,6 +50,15 @@ unwind_frame(struct thread *td, struct unwind_state *frame)
 	if (!kstack_contains(td, fp - sizeof(fp) * 2, sizeof(fp) * 2))
 		return (false);
 
+#ifdef __CHERI_PURE_CAPABILITY__
+	if ((ptraddr_t)(fp - sizeof(fp) * 2) < cheri_getbase(fp) ||
```

Comment:
Does it make sense to add this check to kstack_contains or do we also use that for invalid pointers?

Reply:
kstack_contains uses the td_kstack members for its bounds check which may or may not match the bounds of `fp` itself (which is the thing we are about to dereference).  In particular, if we have a stack with an unexpected frame (which is the case that I ran into) you might have a `fp` cap that is within the bounds of td_kstack, but can't be dereferenced (e.g. a spilled bounded stack pointer or some such).